pallets/lbp/src/lib.rs

```diff
@@ -341,6 +344,17 @@ pub mod pallet {
 	pub type PoolData<T: Config> =
 		StorageMap<_, Blake2_128Concat, PoolId<T>, Pool<T::AccountId, T::BlockNumber>, OptionQuery>;
 
+	/// Storage used for tracking existing fee collectors
+	/// Not more than one fee collector per asset possible
+	#[pallet::storage]
+	#[pallet::getter(fn fee_collectors)]
+	pub type FeeCollectors<T: Config> =
+		StorageDoubleMap<_,
+			Blake2_128Concat,
+			T::AccountId,
+			Blake2_128Concat,
+			AssetId, bool, OptionQuery>;
```

Comment:
Just a note to be aware : you get the default value of bool if such entry does not exist in the map. In case of bool - it is false, that's probably what you want.

however, as i can see - you dont use the get by value at all anyway.